pkg/synthetictests/allowedalerts/basic_alert.go

```diff
@@ -0,0 +1,303 @@
+package allowedalerts
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/origin/pkg/test/ginkgo"
+
+	testresult "github.com/openshift/origin/pkg/test/ginkgo/result"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/openshift/origin/pkg/synthetictests/platformidentification"
+
+	"github.com/openshift/origin/pkg/monitor"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+type AlertTest interface {
+	// TestNamePrefix is name for this as a late test
+	TestNamePrefix() string
+	// LateTestNameSuffix is name for this as a late test
+	LateTestNameSuffix() string
+	// InvariantTestName is name for this as an invariant test
+	InvariantTestName() string
+
+	// AlertName is the name of the alert
+	AlertName() string
+	// AlertState is the threshold this test applies to.
+	AlertState() AlertState
+
+	TestAlert(ctx context.Context, restConfig *rest.Config) error
+	InvariantCheck(ctx context.Context, restConfig *rest.Config, intervals monitorapi.Intervals) ([]*ginkgo.JUnitTestCase, error)
+
+	// FailAfter is the amount of time that an alert can be at or above the current state before failing a test
+	FailAfter(jobType platformidentification.JobType) time.Duration
+	// FlakeAfter is the amount of time that an alert can be at or above the current state before flaking a test
+	FlakeAfter(jobType platformidentification.JobType) time.Duration
+}
+
+// AlertState is the state of the alert. They are logically ordered, so if a test says it limits on "pending", then
+// any state above pending (like info or warning) will cause the test to fail.
+type AlertState string
```

Comment:
Would it be helpful to have these map to integers for a programatic way to compare if a state is above your threshold, instead of having to hard code those assumptions when used? Seems like someone may miss that distinction when using at some point especially with info after pending, and this may be error prone to maintain. Those assumptions appear in couple places in this PR alone. 

Alternatively an AtOrAboveAlertStateThreshold func on the interface might do the trick.

Reply:
Where would such an integer be used in this PR?

Reply:
I was thinking in the switches/filtering where we take level X and above, but if this doesn't trigger any ideas that would be an improvement then disregard.